src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -706,7 +721,27 @@ public static void onWorldLoad(@NotNull final World world)
                 if (data != null)
                 {
                     readFromNBT(data, world);
+
+                    if (data.hasKey(TAG_NEW_COLONIES))
+                    {
+                        int size = data.getInteger(TAG_NEW_COLONIES);
+
+                        @NotNull final File saveDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), FILENAME_MINECOLONIES_PATH);
+                        for (int colonyId = 0; colonyId <= size; colonyId++)
+                        {
+                            @Nullable final NBTTagCompound colonyData = loadNBTFromPath(new File(saveDir, String.format(FILENAME_COLONY, colonyId)));
+                            if (colonyData != null)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)